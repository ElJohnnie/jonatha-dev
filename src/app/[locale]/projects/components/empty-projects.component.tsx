export default function EmptyBlog({ text }: Readonly<{ text: string }>) {
  return (
    <section className='flex h-full flex-col items-center justify-center'>
      <div className='mx-auto flex text-center lg:max-w-none'>
        <h1 className='mb-4 text-3xl font-extrabold leading-none tracking-tight md:text-6xl dark:text-white'>
          {text}
        </h1>
      </div>
    </section>
  );
}
