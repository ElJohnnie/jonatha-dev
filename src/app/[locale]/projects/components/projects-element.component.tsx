import { ProjectElementComponent } from '../types';
import Link from 'next/link';
import Image from 'next/image';
import { useLocale } from 'next-intl';

export default function ProjectElement({
  project,
}: Readonly<ProjectElementComponent>) {
  const locale = useLocale();
  return (
    <>
      <article
        className='group relative h-64 w-full overflow-hidden rounded-lg bg-cover bg-center shadow-lg transition  duration-300 ease-in-out hover:shadow-2xl'
        style={{
          backgroundImage:
            "url('https://images.unsplash.com/photo-1623479322729-28b25c16b011?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1740&q=80')",
        }}
      >
        <div className='absolute inset-0 bg-black bg-opacity-50 transition duration-300 ease-in-out group-hover:opacity-75'></div>
        <div className='relative flex h-full w-full items-center justify-center px-4 sm:px-6 lg:px-4'>
          <h3 className='text-center'>
            <Link
              className='text-center text-2xl font-bold text-white'
              href={`/${locale}/projects/${project.slug}`}
            >
              <span className='absolute inset-0'></span>
              {project.title}
            </Link>
          </h3>
        </div>
      </article>
    </>
  );
}
