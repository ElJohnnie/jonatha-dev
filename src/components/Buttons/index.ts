import AnimateButton from './AnimateButton';

export { AnimateButton };
