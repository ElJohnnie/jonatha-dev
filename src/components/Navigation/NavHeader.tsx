'use client';
import { useMemo } from 'react';
import { useTranslations } from 'next-intl';
import Image from 'next/image';
import Logo from '../../../public/JF.png';
import { usePathname } from 'next/navigation';
import Link from 'next/link';

function classNames(...classes: any) {
  return classes.filter(Boolean).join(' ');
}

export default function NavHeader() {
  const t = useTranslations('Header');
  const pathname = usePathname();

  const links = useMemo(() => {
    return [
      {
        label: t('home'),
        href: '/',
        current: pathname === '/',
      },
      {
        label: t('about'),
        href: '/about',
        current: pathname === '/about',
      },
      // {
      //   label: t('projects'),
      //   href: '/projects',
      //   current: pathname === '/projects',
      // },
    ];
  }, [t, pathname]);

  return (
    <header className='absolute top-0 z-10 order-1 mt-3 flex min-h-[60px] w-full'>
      <nav className='mt-4 flex-1 px-5 min-[480px]:order-2 min-[480px]:mt-0 '>
        <ul className='flex flex-row items-center justify-normal gap-4'>
          <Link
            href='/'
            className='text-primary pl-5 text-lg font-bold sm:mr-5 lg:mr-12'
          >
            <Image src={Logo} alt='Your Company' width={45} height={45} />
          </Link>
          {links.map((item) => (
            <Link
              key={item.label}
              href={item.href}
              className={classNames(
                item.current
                  ? 'bg-gray-900 text-white'
                  : 'text-gray-300 hover:bg-gray-700 hover:text-white',
                'rounded-md px-3 py-2 text-sm font-medium'
              )}
              aria-current={item.current ? 'page' : undefined}
            >
              {item.label}
            </Link>
          ))}
        </ul>
      </nav>
    </header>
  );
}
