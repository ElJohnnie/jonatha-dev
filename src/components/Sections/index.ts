import Hero from './Hero/Hero';
import About from './About/About';
import Blog from './Blog/Blog';

export { Hero, About, Blog };
