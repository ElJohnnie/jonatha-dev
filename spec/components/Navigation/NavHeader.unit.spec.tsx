import React from 'react';
import { NavHeader } from '../../../src/components/Navigation';
import { render } from '@testing-library/react';
import { NextIntlClientProvider } from 'next-intl';

jest.mock('next-intl', () => ({
  ...jest.requireActual('next-intl'),
  useTranslation: jest.fn().mockReturnValue({ t: (key: any) => key }),
}));

let messages: any;

describe('should render correctly NavHeader component', () => {
  it('match snapshot', () => {
    const { container } = render(
      <NextIntlClientProvider locale='en' messages={messages}>
        <NavHeader locale='pt' />
      </NextIntlClientProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
