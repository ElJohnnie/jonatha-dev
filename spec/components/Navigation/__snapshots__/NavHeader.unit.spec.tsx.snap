// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly NavHeader component match snapshot 1`] = `
<header
  class="absolute top-0 z-10 order-1 mt-3 flex min-h-[60px] w-full"
>
  <nav
    class="mt-4 flex-1 px-5 min-[480px]:order-2 min-[480px]:mt-0 "
  >
    <ul
      class="flex flex-row items-center justify-normal gap-4"
    >
      <a
        class="text-primary pl-5 text-lg font-bold sm:mr-5 lg:mr-12"
        href="/"
      >
        <img
          alt="Your Company"
          data-nimg="1"
          decoding="async"
          height="45"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="45"
        />
      </a>
      <a
        class="text-gray-300 hover:bg-indigo-950 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
        href="/"
      >
        Header.home
      </a>
      <a
        class="text-gray-300 hover:bg-indigo-950 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
        href="/about"
      >
        Header.about
      </a>
    </ul>
  </nav>
</header>
`;
